src/Apps/ArtQuiz/artQuizRoutes.tsx

```diff
@@ -1,8 +1,26 @@
 import loadable from "@loadable/component"
-import { RedirectException } from "found"
+import { Match, RedirectException } from "found"
 import { graphql } from "react-relay"
+import { ArtsyResponse } from "Server/middleware/artsyExpress"
 import { AppRouteConfig } from "System/Router/Route"
 
+const LOGIN_COPY = "Log in to take the Art Quiz."
+const REDIRECT_URL = `/login?redirectTo=/art-quiz&afterSignUpAction=/art-quiz&copy=${LOGIN_COPY}`
+
+const artQuizServerSideRedirect = ({ res }: { res: ArtsyResponse }) => {
+  if (!res.locals.sd.FEATURE_FLAGS["art-quiz"].flagEnabled) {
+    res.redirect("/")
+  }
+  if (!res.locals.sd.CURRENT_USER) {
+    res.redirect(REDIRECT_URL)
+  }
+}
+const artQuizClientSideRedirect = ({ match }: { match: Match }) => {
```

Comment:
Why would we need a client-side redirect?

Reply:
The assumption was that this would be relevant if the user logged out mid quiz or something, but I didn't think too deeply about it. Probably incorrect?

Reply:
Yeah, safe to remove this. Logging out does a hard-refresh anyway.

Reply:
done